doc/md/declarative/declarative-hcl.mdx

```diff
@@ -0,0 +1,225 @@
+
+### Usage:
+
+```console
+atlas schema apply [flags]
+```
+
+### Example:
+
+Let’s apply the changes we created in Step 2 using the following command:
+
+```console
+atlas schema apply \
+  --url "mysql://root:pass@localhost:3306/example" \
+  --to "file://schema.hcl"
+```
+
+Atlas compares  the current state of the database with the state described in the schema file, and shows you any necessary changes to bring the database in line with the schema definition.
+
+```console
+-- Planned Changes:
+-- Create "orders" table
+CREATE TABLE `example`.`orders` (`id` int NOT NULL AUTO_INCREMENT, `customer_id` int NOT NULL, `order_date` date NOT NULL, PRIMARY KEY (`id`), INDEX `orders_ibfk_1` (`customer_id`), CONSTRAINT `orders_ibfk_1` FOREIGN KEY (`customer_id`) REFERENCES `example`.`customers` (`id`) ON UPDATE NO ACTION ON DELETE NO ACTION) CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci;
+Use the arrow keys to navigate: ↓ ↑ → ← 
+? Are you sure?: 
+  ▸ Apply
+    Abort
+```
+
+Selecting `Apply` will apply the changes to the local environment.
+
+You can inspect the latest state of the database into the schema.hcl file again and verify that the changes have been applied by re-running the `atlas schema inspect` command:
+
+```console
+atlas schema inspect -u "mysql://root:pass@localhost:3306/example"
+```
+```hcl
+table "customers" {
+  schema = schema.example
+  column "id" {
+    null           = false
+    type           = int
+    auto_increment = true
+  }
+  column "first_name" {
+    null = false
+    type = varchar(255)
+  }
+  column "last_name" {
+    null = false
+    type = varchar(255)
+  }
+  primary_key {
+    columns = [column.id]
+  }
+}
+table "orders" {
+  schema = schema.example
+  column "id" {
+    null           = false
+    type           = int
+    auto_increment = true
+  }
+  column "customer_id" {
+    null = false
+    type = int
+  }
+  column "order_date" {
+    null = false
+    type = date
+  }
+  primary_key {
+    columns = [column.id]
+  }
+  foreign_key "orders_ibfk_1" {
+    columns     = [column.customer_id]
+    ref_columns = [table.customers.column.id]
+    on_update   = NO_ACTION
+    on_delete   = NO_ACTION
+  }
+  index "orders_ibfk_1" {
+    columns = [column.customer_id]
+  }
+}
+schema "example" {
+  charset = "utf8mb4"
+  collate = "utf8mb4_0900_ai_ci"
+}
+```
+We can confirm that the table has been effectively inserted into the `example` schema.
+
+:::info
+By following the steps above, you can inspect the current state of the database, make changes to the schema definition file and apply those changes to the database with confidence.
+:::
+
+### Conclusion
+
+The declarative migration is a flexible way to manage your database schema. It allows you to define your desired state in a simple, human-readable language, and then automatically apply those changes to your database. This can save you time and effort, and also reduce the risk of errors or inconsistencies in your database schema.
```

Comment:
link to conceptual articles for deeper reads: declarative vs versioned, SQL vs hcl. we want to let the user know that atlas can work in other ways as well